spec/system/header_search_spec.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+RSpec.describe "Header Search - Responsive Behavior", type: :system do
+  fab!(:user)
+  fab!(:topics) { Fabricate.times(20, :post).map(&:topic) }
+
+  before do
+    SiteSetting.search_experience = "search_field"
+    sign_in(user)
+  end
+
+  it "appears based on scroll & screen width with search banner enabled" do
+    SiteSetting.enable_welcome_banner = true
+    visit "/"
+
+    # Default desktop view should show search field
+    expect(page).to have_no_css(".floating-search-input")
```

Comment:
We have a page object for search related system spec stuff already https://github.com/discourse/discourse/blob/abee294616a064e4de40d8e8eb052ea571e2203c/spec/system/page_objects/pages/search.rb , you should use this where possible in this spec